src/SDK/SmartDetectorsSDK/TelemetryDataClient/TelemetryReourceNotFoundException.cs

```diff
@@ -0,0 +1,57 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TelemetryReourceNotFoundException.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+namespace Microsoft.Azure.Monitoring.SmartDetectors
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// An exception indicating the telemetry resource was not found
+    /// </summary>
+    [Serializable]
+    public class TelemetryReourceNotFoundException : Exception
```

Comment:
>TelemetryReourceNotFoundException [](start = 17, length = 33)

can you please also add throwing it?
in TelemetryDataClientBase.RunQueryAsync just check for NotFound response

Reply:
actually - there's already handling it - when handling BadRequest

---
In reply to: [258886262](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/51#discussion_r258886262) [](ancestors = 258886262)

Reply:
right now we return null there. I opened a bug on next sprint.

---
In reply to: [258886262](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/51#discussion_r258886262) [](ancestors = 258886262)